db/blip_handler.go

```diff
@@ -306,6 +306,9 @@ func (bh *blipHandler) sendChanges(sender *blip.Sender, opts *sendChangesOptions
 						if change.Revoked {
 							deleted += 2
 						}
+						if len(change.Removed) > 0 {
+							deleted += 4
+						}
```

Comment:
I think I've mentioned this before, but I really dislike this way of setting flags as decimal values in a bitfield. It ultimately just looks like magic numbers, and glancing at this code, it's not obvious that `deleted` is just a set of flags that mean more than just "deleted".

It would be slightly better if:
- `deleted` was renamed to `deletedFlags`
- `1`, `2`, and `4` were named consts (i.e. `changesDeletedFlagRemoved`)

But the `+=` might still throw people off I think.
Adding a new type for the flag consts gives yet more hints that people shouldn't be including magic numbers, and should only be using the defined consts.

```go
type changesDeletedFlag uint

const (
	// could use iota here, but the replication protocol
	// should define them explicitly, so why not here too
	changesDeletedFlagDeleted changesDeletedFlag = 0b001
	changesDeletedFlagRevoked changesDeletedFlag = 0b010
	changesDeletedFlagRemoved changesDeletedFlag = 0b100
)
```

```go
	var deletedFlags changesDeletedFlag
	if change.Deleted {
		deletedFlags |= changesDeletedFlagDeleted
	}
	if change.Revoked {
		deletedFlags |= changesDeletedFlagRevoked
	}
	if change.Removed {
		deletedFlags |= changesDeletedFlagRemoved
	}
```